### ISSUE TITLE
Builds of Positron for arm64 on Windows

### ISSUE THREAD

**Issue by lionel-** [posit-dev/positron#5254]
Requested in https://github.com/posit-dev/positron/discussions/5252, https://github.com/posit-dev/positron/discussions/5249, and by a couple of users in https://github.com/posit-dev/positron/issues/2934.

Depends on building Ark for arm64 Windows: https://github.com/posit-dev/positron/issues/2934.

**Comment by drkamarul** [comment]
Any latest update on getting Positron IDE on arm64 Windows?

**Comment by denis-or** [comment]
I tried installing Positron on my Asus Vivobook using WSL. So yeah, I went ahead and set it up on Ubuntu. But obviously, it didn't work. lol Any updates on this?

**Comment by juliasilge** [comment]
We don't currently have any updates on Windows ARM builds but we appreciate any 👍 or additional info on this issue, to help with prioritization for this work!

**Comment by MoizesCBF** [comment]
Does Positron run in an arm64 Linux PC? If so, which distribution could I install for that?

**Comment by jmcphers** [comment]
Yes it does! You can get a build of Positron for arm64 Linux on our download page. Ubuntu 24 is a good choice for a Linux distribution if you don't have other constraints. https://positron.posit.co/download.html

**Comment by juliasilge** [comment]
Request in discussions at https://github.com/posit-dev/positron/discussions/6945

**Comment by gdickens** [comment]
Just adding my vote for Positron for ARM64.

**Comment by ProfLear** [comment]
Yep!  I would love this.  I think positron is amazing, and being able to use it on the snapdragon chip I have on my laptop would be great.

**Comment by mickeymaples** [comment]
Yeah I would love an ARM version for windows

**Comment by jackgidding** [comment]
Really need arm64 for Windows. I've been using Mac / Apple Silicon version but need to switch to a Windows device. I've had to switch back to Jupyter.

**Comment by ProfLear** [comment]
> had to switch back to Jupyter

If you are going that way, you might consider [Marimo](https://marimo.io/), which I find to be a much better implementation of the notebook idea.

**Comment by digbyphonic** [comment]
Am interested in an ARM64 version of Positron.

Loyal RStudio / Quarto user on Windows 11 x64. 
On second machine with ARM64 I've installed WSL2 to try and run RStudio / Quarto; but, it's not satisfactory.
Would be interested in hearing if the Ubuntu version of Positron works well in WSL2?

Happy to embrace the future and make the switch to Positron, if that's recommended?
An ARM64 version would seal the deal.

Thank you for the good work. Exciting stuff.

**Comment by juliasilge** [comment]
Requested on one of our YouTube videos [here](https://www.youtube.com/watch?v=4Ir_HX4riHw&lc=Ugx3IEAsKQNaHb0H-5h4AaABAg)

**Comment by digbyphonic** [comment]
> Am interested in an ARM64 version of Positron.
> 
> Loyal RStudio / Quarto user on Windows 11 x64. On second machine with ARM64 I've installed WSL2 to try and run RStudio / Quarto; but, it's not satisfactory. Would be interested in hearing if the Ubuntu version of Positron works well in WSL2?
> 
> Happy to embrace the future and make the switch to Positron, if that's recommended? An ARM64 version would seal the deal.
> 
> Thank you for the good work. Exciting stuff.

I should add that my current workaround uses R and Quarto in VS Code within standard Windows 11 ARM64.
https://www.r-project.org/nosvn/winutf8/aarch64/R-4-signed/
https://cran.r-project.org/bin/windows/Rtools/rtools44/rtools.html

If Positron on Windows ARM64 were available it would be a better fit.

**Comment by rafaelguariento** [comment]
I am using a Dell Inspiron 14 Plus with a Snapdragon X Plus. It is good machine overall, until it is not, and the problem is that used to use Rstudio a lot, which does not support ARM on windows. Now, I found out that Positron does not support it as well. So, I switched to VS Code and I use the R version for ARM64 on windows (which is not visible do download on the main download page for R but can be installed with the links they provided in their blog post).  So, for everyone struggling with R development (specially doing stats for academic pourpouses) my recommendation is to switch to VS Code. But I am just adding my vote for Positron for ARM64 on Windows.

**Comment by ellisp** [comment]
Just here to add my vote to this. Was very surprised to find I cannot run Positron on a new Surface Pro with snapdragon.

**Comment by hughesan** [comment]
Chiming in as another person with Snapdragon/ARM64 wanting to try Positron!

**Comment by juliasilge** [comment]
Requested again in https://github.com/posit-dev/positron/discussions/9639

**Comment by rafaelguariento** [comment]
There is another issue, this time not related to Positron. I tried installing RcppParallel using the ARM64 version of R, but it failed due to a missing library. Therefore, WSL seems to be the best option for R development on Windows machines with an ARM chip.

**Comment by benediktschfr** [comment]
> There is another issue, this time not related to Positron. I tried installing RcppParallel using the ARM64 version of R, but it failed due to a missing library. Therefore, WSL seems to be the best option for R development on Windows machines with an ARM chip.

It's off-topic, but you may try installing the RcppParallel development version directly from GitHub with 

```R
pak::pak("RcppCore/RcppParallel")
```

**Comment by denis-or** [comment]
> There is another issue, this time not related to Positron. I tried installing RcppParallel using the ARM64 version of R, but it failed due to a missing library. Therefore, WSL seems to be the best option for R development on Windows machines with an ARM chip.

I tried installing Positron on WSL, but I got an error saying it had to be installed on the host operating system..

**Comment by juliasilge** [comment]
We've got Kallichore now! https://github.com/posit-dev/kallichore/pull/51 ✅ 

I think next is ark as tracked in https://github.com/posit-dev/positron/issues/2934.

**Comment by juliasilge** [comment]
We now have ark builds for Windows ARM! https://github.com/posit-dev/ark/pull/940 🎉 

We do need some of the r-lib tools like GH actions and rig and such to be updated to work with the Windows ARM GitHub runners for them to be usable in Positron. I think that's the next step.

**Comment by drkamarul** [comment]
> > Am interested in an ARM64 version of Positron.
> > Loyal RStudio / Quarto user on Windows 11 x64. On second machine with ARM64 I've installed WSL2 to try and run RStudio / Quarto; but, it's not satisfactory. Would be interested in hearing if the Ubuntu version of Positron works well in WSL2?
> > Happy to embrace the future and make the switch to Positron, if that's recommended? An ARM64 version would seal the deal.
> > Thank you for the good work. Exciting stuff.
> 
> I should add that my current workaround uses R and Quarto in VS Code within standard Windows 11 ARM64. https://www.r-project.org/nosvn/winutf8/aarch64/R-4-signed/ https://cran.r-project.org/bin/windows/Rtools/rtools44/rtools.html
> 
> If Positron on Windows ARM64 were available it would be a better fit.

Can you explain how to set up R aarch64 in VS Code? I have downloaded the aarch64 from the link you mentioned. I am currently using R 4.5.1 x86_64-w64-mingw32, which I installed through the VS Code Extension, but it's not the aarch64 version.

**Comment by digbyphonic** [comment]
R on 64-bit ARM Windows - The R Blog<https://blog.r-project.org/2024/04/23/r-on-64-bit-arm-windows/index.html>

R in Visual Studio Code<https://code.visualstudio.com/docs/languages/r>

//
Jonathan J Digby

**Comment by vaulot** [comment]
Voting for ARM64 support. I really do not understand if Posit wants Positron to be adopted, this is still missing.
I have a surface as a second computer and this implementation is badly needed.

**Comment by benediktschfr** [comment]
> Voting for ARM64 support. I really do not understand if Posit wants Positron to be adopted, this is still missing. I have a surface as a second computer and this implementation is badly needed.

I just wanted to jump in here to express my appreciation for the great work the Positron team is doing.

The desire for a native ARM64 version for Windows is completely understandable, especially with the new devices coming out. From what I've read in the developers' comments here, it seems this topic already has a certain priority, which is fantastic to see.

I think it's important to remember that Windows on ARM, particularly with the new Snapdragon processors in the latest Surface laptops, is still a very young and rapidly evolving ecosystem. Porting an application to a new architecture is often much more complex than a simple "recompile." It involves verifying dependencies, adapting toolchains, and thoroughly testing everything on new hardware.

Therefore, I think it's completely reasonable that this kind of fundamental development takes time. Comparing it to other, sometimes much larger, projects isn't always a fair comparison.

A big thank you to the team for having this on your radar! I'm happy to wait patiently for a stable release.

**Comment by karltk** [comment]
Another vote for a ARM64/Windows release. Keep up the good work!

**Comment by jmcphers** [comment]
Hi all, I'm pleased to announce that we have started producing **experimental** builds of Positron for Windows ARM64. If you have time to try them we'd very much appreciate your feedback. You can get an early build here:

System install - https://cdn.posit.co/positron/dailies/win/arm64/Positron-2025.12.0-79-Setup-arm64.exe

User install -  https://cdn.posit.co/positron/dailies/win/arm64/Positron-2025.12.0-79-UserSetup-arm64.exe

Some important notes:

- Currently, only the arm64 version of Python works with this arm64 build of Positron. You'll see errors if you attempt to use the x64 version of Python. See this issue for details: https://github.com/posit-dev/positron/issues/10426
- You can use either the [aarch64 version of R](https://www.r-project.org/nosvn/winutf8/aarch64/) or x64 version of R. Note that if you use the aarch64 version of R, you will have increased performance but won't be able to install most packages since CRAN does not yet publish binaries for aarch64. If you use aarch64 R, install the [RTools for 64-bit ARM](https://cran.r-project.org/bin/windows/Rtools/rtools45/rtools.html) so packages can be compiled locally.
- Positron does not bundle Quarto because Quarto itself does not have a native Windows ARM64 build. If Quarto is essential to your workflow, you may have success installing the x64 version of Quarto, which will run under emulation.

Please ensure auto update is set to the Daily channel so that you can receive updates to this build.

We hope to release arm64 support as an experimental feature in the next monthly release of Positron. Thanks for the support!

**Comment by drkamarul2** [comment]
I just installed it. 

Works flawlessly at the moment.
Connected to GitHub.
Managed to render a quarto presentation as well.
Added Claude extension.  
Have not tried Python codes yet. 

My specs:
R version 4.5.1 [Platform: x86_64-w64-mingw32/x64]

Thanks everyone. I am super excited.

**Comment by ZacWarham** [comment]
Just installed at the system level for a quick look. Wasn't an immediate success but mostly user error I suspect as this is my first time using Positron. Running the x86_64-w64-mingw32/x64 version of R 4.5.1

First thing I did was go and set the default terminal in Positron to be my WSL instance and then I opened a folder to a Shiny App project on a file server. Clicking run app at the top failed as it was trying to launch from the WSL terminal using the file path to my Windows installation of R - is there a way to tell it which terminal to use so I can keep my WSL terminal default but launch with the correct one?

After that I changed back to the "null" default for Windows terminal and launching the app from my file server worked although I saw a lot of these warnings repeating over and over. App still functioned but these did not seem to stop repeating in the logs.

```
Warning in file.info(files) :
  cannot resolve owner of file 'app.R': No mapping between account names and security IDs was done
Warning in file.info(files) :
  cannot resolve owner of file 'backup.R': No mapping between account names and security IDs was done
Warning in file.info(files) :
  cannot resolve owner of file 'v1.R': No mapping between account names and security IDs was done
Warning in file.info(files) :
  cannot resolve owner of file 'www/custom.css': No mapping between account names and security IDs was done
```

First comment is definitely a user error I feel but if anyone has a suggestion of a solution that would be appreciated (or telling me if this is just a silly way to use it 😅). Not sure what to make of the second. Copying the project locally stopped the warnings but I typically work from the server.

And unrelated to this issue, what is the recommended way to stop a Shiny app in Positron? I tried the "Clear current URL" button but the session kept running in the terminal. Is there anything besides `Ctrl` + `C` or the trash on the terminal? I expected a Stop button where the play button was or in the drop down

**Comment by rafaelguariento** [comment]
Just installed, using R for aarch64, set up Copilot pro in Positron Assistant. Everything working with zero problems.

**Comment by jmcphers** [comment]
@ZacWarham I don't think any of those errors are related to the arm64 build! If you prefer to work inside WSL, have you tried connecting to WSL as a remote? Use the _Remote-WSL: Connect to WSL_ command. This will run Positron inside WSL so you can interact with WSL files and terminals more natively. You'll also see Linux versions of Python and R that are installed inside your WSL Linux distribution instead of the Windows versions.

<img width="664" height="125" alt="Image" src="https://github.com/user-attachments/assets/e60aaa41-5d88-4dee-8846-0c5008ec1e3c" />

The problem you're describing with stopping Shiny apps is a known issue! You can comment/upvote here: https://github.com/posit-dev/positron/issues/7747 (see also https://github.com/posit-dev/positron/issues/5344).

**Comment by ZacWarham** [comment]
Glad to hear it is user error and not arm64 issues! 

> Use the Remote-WSL: Connect to WSL command. 

Thank you for the recommendation. I will be using this normally but our students will not be so I was testing two methods at the same time 😅

> stopping Shiny apps is a known issue

I thought it must be. Thanks for pointing me in the right direction!

**Comment by gdickens** [comment]
Works for me on my Surface Pro 11. 

Thanks!

**Comment by denis-or** [comment]
Hi everyone!

Great news! 

I installed it here and everything is fine so far. I imported my VS Code configuration without any problems.

My specs too:
R 4.5.2 aarch64-w64-mingw32 and R x86_x64 too
RTools 4.5 aarch64
Rtools 4.5 x64
Windows 11 Home 
ASUS Vivobook S with Snapdragon Elite X

**Comment by benediktschfr** [comment]
Just wanted to chime in and report that I've installed Positron as well and haven't faced any issues so far. It's been working smoothly for me.

**My current workflows in Positron include:**
* Creating Quarto documents
* Standard R workflows
* R Package development and maintenance

**Regarding R CMD Check:**
The only error I encounter during package checks is the following:

```text
ERROR: Unknown command "TMPDIR=C:/Users/bened/AppData/Local/Temp/RtmpQRQBuo/file36a445927ce0". Did you mean command "render"?
Warning message:
In system2("quarto", "-V", stdout = TRUE, env = paste0("TMPDIR=",  :        
   running command '"quarto" TMPDIR=C:/Users/bened/AppData/Local/Temp/RtmpQRQBuo/file36a445927ce0 -V' had status 1
```

However, as mentioned in the discussion regarding the dev version, this seems to be related to my specific Quarto installation/setup rather than a Positron bug. Aside from this log message, the checks actually pass correctly.

<details>
<summary>My System Information</summary>

```r
R version 4.5.2 (2025-10-31 ucrt)
Platform: aarch64-w64-mingw32
Running under: Windows 11 (build 26200)

Matrix products: default
  LAPACK version 3.12.1

locale:
[1] LC_COLLATE=German_Germany.utf8  LC_CTYPE=German_Germany.utf8    LC_MONETARY=German_Germany.utf8 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.5.2 tools_4.5.2   
```

</details>

**Comment by denis-or** [comment]
> Hi everyone!
> 
> Great news!
> 
> I installed it here and everything is fine so far. I imported my VS Code configuration without any problems.
> 
> My specs too: R 4.5.2 aarch64-w64-mingw32 and R x86_x64 too RTools 4.5 aarch64 Rtools 4.5 x64 Windows 11 Home ASUS Vivobook S with Snapdragon Elite X

Update:
I had a problem when trying to plot a graph, whether using ggplot or just plot. It was the same problem reported in #3965, and I solved it by changing the r_profile as well.

**Comment by vaulot** [comment]
Thanks a lot. I have installed it on my Surface and it is running without any trouble.

**Comment by ZacWarham** [comment]
Just came across an issue but not sure if it is related so I created a separate issue here (I do not have a separate machine to test on) - https://github.com/posit-dev/positron/issues/10811

Edit: it was not related

**Comment by digbyphonic** [comment]
Excellent work.

Used for more than a week; has worked flawlessly so far.

Surface Pro ARM 64, Positron with R-aarch64\R-4.5.0

Thank you,  team Positron!

**Comment by jmcphers** [comment]
Hi all, thank you so much for being willing to test early builds! We've now made these builds available (albeit in an experimental form) in our latest release: https://github.com/posit-dev/positron/releases/tag/2025.12.0-167

**Comment by rodrigosf672** [comment]
### Verified Fixed
    Positron Version(s) : 2025.12.0 build 167
    OS Version          : Surface Pro ARM 64

- Closing this as verified based on the overwhelmingly positive response from users (thank you!). 
- Users were able to download and use Positron without any issues so far. 
- If issues specific to Surface Pro arm64 builds emerge, please feel free to log issues so that they can be triaged in the future.
